File: RiskAssessment/API/RiskManagement.Web/src/app/pages/assignment/assignment.model.ts
//import { NgbModal, NgbDateStruct, NgbTimeStruct } from '@ng-bootstrap/ng-bootstrap';



export class Assignment {
    public AssignmentId: number;
    public Title: string;
    public Active: boolean;
    ///public OrganizationId: number;
}

export class AssignmentQuestionOption {
    public QuestionOptionID: number;
    public QuestionID: number;
    public OptionText: string;
    public IsCorrectAnswer: boolean;
}

export class AssignmentListResult {
    public QuestionId: number;
    public AssignmentId: number;
    public QuestionText: string;
    public QuestionType: string;
    public Score: number;
    public Order: number;
    public Active: boolean = true;
    public AssignmentQuestionOptionViewModel: AssignmentQuestionOption[];
}

export class AssignmentListFilterModel {
    public Title: string;
    public Active: boolean;
    public PageNo: number;
    public PageSize: number;
    public SortOrder: string;
    public SortColumn: string;
}



export class AssignmentListData {

    public TotalRecords: number;
    public AssignmentListResult: AssignmentListResult[] = [];
}

export class AssignmentListResultV1 {
    public Title: string
    public RowID: number;
    public AssignmentID: number;
    public Active: string;
}
export class AssignmentListViewResult {

    public TotalRecords: number;
    public AssignmentListResult: AssignmentListResultV1[] = [];
}

File: Angular2_AspDotNet/API/Angular2_AspDotNet.Models/AssignmentModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Angular2_AspDotNet.Models
{
    public class AssignmentListFilterModel
    {
        public int AssignmentID { get; set; }
        public string Title { get; set; }
        public bool Active { get; set; }
        public int PageNo { get; set; }
        public int PageSize { get; set; }
        public string SortOrder { get; set; }
        public string SortColumn { get; set; }
    }
    public class AssignmentListData
    {
        public AssignmentListData()
        {
            this.AssignmentListQuestionResult = new List<Models.AssignmentListQuestionResult>();
        }
        public int TotalRecords { get; set; }
        public List<AssignmentListQuestionResult> AssignmentListQuestionResult { get; set; }
    }
    public class AssignmentListResult
    {
        public string Title { get; set; }
        public int AssignmentID { get; set; }
        public Nullable<long> RowID { get; set; }
        public string Active { get; set; }
    }
    public class AssignmentListViewResult
    {
        public AssignmentListViewResult()
        {
            this.AssignmentListResult = new List<AssignmentListResult>();
        }
        public int TotalRecords { get; set; }
        public List<AssignmentListResult> AssignmentListResult { get; set; }
    }

    public class AssignmentQuestionListViewModel
    {
        public int AssignmentID { get; set; }
        public string QuestionText { get; set; }
        public Nullable<int> Score { get; set; }
        public Nullable<int> Order { get; set; }
        public string MultipleOptionSet { get; set; }
        public string QuestionType { get; set; }
        public Nullable<int> QuestionID { get; set; }
    }

}
